tests/tape/tapes/test_qnode.py

```diff
@@ -600,3 +601,31 @@ def func(x, y):
 
         assert np.allclose(res, res2, atol=tol, rtol=0)
         assert func.qtape is not old_tape
+
+
+class TestQNodeCollection:
+    """Unittests for the QNodeCollection"""
+
+    def test_multi_thread(self):
+        n_qubits = 4
+        n_batches = 5
+        dev = qml.device("default.qubit", wires=n_qubits)
+
+        qml.enable_tape()

```

Comment:
Hey @shukob! I think the reason some random tests are failing is that tape mode is never disabled after this test, so subsequent tests are being run in tape mode (when they shouldn't).

To fix this, it is best to create a fixture that automates the process of turning tape mode on and off for this test, _even if the test fails_. This can be done like so:

```python
class TestQNodeCollection:
    """Unittests for the QNodeCollection"""

    @pytest.fixture
    def enable_tape_mode(self):
        qml.enable_tape()
        yield
        qml.disable_tape()

    def test_multi_thread(self, enable_tape_mode):
        n_qubits = 4
        n_batches = 5
        dev = qml.device("default.qubit", wires=n_qubits)

        def circuit(inputs, weights):
```

The `enable_tape_mode` fixture will make sure that the `test_multi_thread` test will always tear down tape mode when complete.

Reply:
@josh146 Thank you! I mistakenly thought the failure was caused by another commit in anotherbranch. Thank you for providing the solution to the failure.